src/components/tl/ucc_tl.c

```diff
@@ -242,6 +242,11 @@ ucc_status_t ucc_tl_team_create_multiple(ucc_team_multiple_req_t *req)
     }
     req->descs[*id].status = UCC_TL_CTX_IFACE(req->descs[*id].ctx)
                                ->team.create_test(&req->descs[*id].team->super);
+    if (req->descs[*id].status < 0) {
+        /* if team create failed in team create test need to clenup resources */
+        UCC_TL_CTX_IFACE(req->descs[*id].ctx)->team.destroy(
```

Comment:
>Why isn't the responsability of team_create_test to cleanup the team in case of failure ?

This our internal agreement, see here https://github.com/openucx/ucc/issues/341